btf/btf.go

```diff
@@ -680,10 +680,6 @@ func newHandleFromRawBTF(btf []byte) (*Handle, error) {
 		return &Handle{fd, attr.BtfSize}, nil
 	}
 
-	if err := haveBTF(); err != nil {
```

Comment:
Hmm, not sure we should really yank this. It's nice to get ErrNotSupported as a user, as can be seem from the tests that manually call feature probes to skip the test.

Why not redefine `haveBTF` to mean "when BPF_LOAD_BTF" was added by dropping the KIND_VAR? You can still introduce `haveMapBTF` and use it from MarshalKV.

Optionally: call `haveBTF` from `NewHandle` instead of `newHandleFromRawBTF`. That punts the problem to when we get rid of `Marshal*` at some point.

Reply:
> Why not redefine haveBTF to mean "when BPF_LOAD_BTF" was added by dropping the KIND_VAR? You can still introduce haveMapBTF and use it from MarshalKV.

That's what I had on my first attempt at this patch (load a simple blob with an int), but it didn't address the problem of a map with int/int kv succeeding to load the BTF blob, but the map type not supporting BTF metadata. I only added the map load retry later.

I just realized removing `haveBTF()` here breaks the API contract of `NewHandle()`, which promises to return ErrNotSupported. I'll reintroduce a `haveBTF()` that loads an int.

> Optionally: call haveBTF from NewHandle

I think `newHandleFromRawBTF` is exactly where it belongs. These probes are actually quite cheap, and the more often they're called, the better they amortize. It's good to have accurate feedback from building blocks like these, that makes the logic on top more solid and less complicated.